app/controllers/leaderboards_controller.rb

```diff
@@ -0,0 +1,36 @@
+class LeaderboardsController < ApplicationController
+  before_action :set_quiz
+
+  def index
+    @entries = @quiz.leaderboard_entries
+                    .select("name, MAX(score) AS score")
+                    .group(:name)
+                    .order("score DESC")
+                    .limit(10)
+  end
+
+  def create
+    answers = session[:quiz_answers]&.[](@quiz.id.to_s) || {}
+    score = @quiz.questions.sum { |q| answers[q.id.to_s] == q.correct_option ? 1 : 0 }
```

Comment:
fine for now, but i wonder if this should be a method on the quiz model which is unit tested. i.e. 

```
score = @quiz.calculate_score(answers)
```

Also it might be a bit more readable to use fetch 

```
answers = session.fetch(:quiz_answers, {})&.fetch(@quiz_id.to_s, {})
```

or you can use dig 

`answers = session.dig(:quiz_answers, @quiz_id.to_s) || {}`